flink-python/src/main/java/org/apache/flink/python/env/beam/ProcessPythonEnvironmentManager.java

```diff
@@ -430,4 +442,117 @@ private static String createBaseDirectory(String[] tmpDirectories) throws IOExce
                         + Arrays.toString(tmpDirectories)
                         + "' for storing the generated files of python dependency.");
     }
+
+    private static final class PythonEnvResources {
+
+        private static final ReentrantLock lock = new ReentrantLock();
+
+        @GuardedBy("lock")
+        private static final Map<JobID, PythonLeasedResource> reservedResources = new HashMap<>();
```

Comment:
```suggestion
        private static final Map<Object, PythonLeasedResource> reservedResources = new HashMap<>();
```

It doesn't necessarily to be JobID and we just happen to use JobID. So it should be enough to declare it as Object type.